draft-ietf-quic-transport.md

```diff
@@ -1069,6 +1069,18 @@ to cease using the connection IDs when requested can result in connection
 failures, as the issuing endpoint might be unable to continue using the
 connection IDs with the active connection.
 
+An endpoint MAY elect to only send or retransmit RETIRE_CONNECTION_ID frames
+with sequence numbers greater than or equal to the highest Retire Prior To field
+received minus its advertised active_connection_id_limit. This bounds the
+necessary state at the endpoint. For example, if an endpoint receives a Retire
+Prior To field of 7 and has an advertised active_connection_id_limit of 4, it
+may cease sending any RETIRE_CONNECTION_ID frame with sequence number less than
```

Comment:
Is this the functionality we want?  If we're going to go this direction, we might as well just say you don't need to send RETIRE_CONNECTION_ID for anything below Retire Prior To, and you can assume it'll be retired 'soon'.  Maybe that's a good solution anyway?

Reply:
The value of RCID is to let the receiver know that it will never receive a packet with that CID again, barring packet reordering. (one could use an ACK of NCID as an implicit signal, though we don't do that in QUIC). I think we want to preserve that. If the language I wrote puts clients in the situation where they're not getting all the RCIDs, it's because they are buggy, malicious, or in a pretty extreme corner case.

Moreover your suggestion is a wider departure from the spec, likely to create more problems for existing implementations.

Reply:
This seems too concrete to me.